### PR TITLE
Fix package imports

### PR DESCRIPTION
**Pull request by Taar**
- Change eslint plugins for linting imports
- Move core code out of the index file and into its own file. This is so
that the index file can be used strictly for exporting
- Add exports for all measures from the index file in definitions
- Update the README to reflect how imports work now

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,8 +2,7 @@
   "name": "convert-units",
   "version": "3.0.0-beta.0",
   "description": "Convert between quantities in different units",
-  "main": "./lib/cjs/index.js",
-  "module": "./lib/esm/index.js",
```
Comment (maneetgoyal) [inline package.json:6]: Do we have to remove the `"module"` field? From what I remember, `"module"` can be used by bundlers like Webpack/Rollup in order to do better (??) bundling. [This](https://github.com/rollup/rollup/wiki/pkg.module) may help.

File: .eslintrc.js [hunk 2]
```diff
@@ -11,7 +11,8 @@ module.exports = {
     'plugin:@typescript-eslint/eslint-recommended',
     'plugin:@typescript-eslint/recommended',
     'prettier',
-    'plugin:ordered-imports/recommended',
+    'plugin:import/recommended',
+    'plugin:import/typescript'
```
[the remark this answers is not in the document]
Comment (Taar) [reply .eslintrc.js]: Yup you're correct, thanks for pointing this out!

File: package.json (hunk 1 above)
Comment (Taar) [reply package.json:6]: Thanks for pointing this out! I meant to circle back and figure out what "module" does before committing my changes.

**Comment by Taar** [comment]
@maneetgoyal Let me know your thoughts on the last commit and if it satisfies your comments/suggestions. Thanks!

**Review by maneetgoyal — APPROVED** [review]
👍

**Pull request merged**